crates/lox_core/src/time/epochs.rs

```diff
@@ -62,6 +63,28 @@ impl Epoch {
         let second = day_in_seconds + hour_in_seconds + minute_in_seconds + time.second();
         let attosecond = time.attosecond();
         let raw = RawEpoch { second, attosecond };
+        Self::from_raw(scale, raw)
+    }
+
+    pub fn from_datetime(scale: TimeScale, dt: DateTime) -> Self {
+        Self::from_date_and_time(scale, dt.date(), dt.time())
+    }
+
+    /// Returns the J2000 epoch in the given timescale.
+    pub fn j2000(scale: TimeScale) -> Self {
+        Self::from_raw(scale, RawEpoch::default())
+    }
+
+    /// Returns, as an epoch in the given timescale, midday on the first day of the proleptic Julian
+    /// calendar.
+    pub fn jd0(scale: TimeScale) -> Self {
+        // This represents 4713 BC, since there is no year 0 separating BC and AD.
+        let first_proleptic_day = Date::new_unchecked(ProlepticJulian, -4712, 1, 1);
+        let midday = Time::new(12, 0, 0).expect("midday should be a valid time");
+        Self::from_date_and_time(scale, first_proleptic_day, midday)
+    }
+
+    fn from_raw(scale: TimeScale, raw: RawEpoch) -> Self {
```

Comment:
I am not entirely convinced about these API changes but we should revisit this later.